File: extensions/intellij/intellij_generator_plugin/src/main/java/com/bloc/intellij_generator_plugin/live_templates/BlocContext.java
package com.bloc.intellij_generator_plugin.live_templates;

import com.intellij.codeInsight.template.TemplateContextType;
import com.intellij.psi.PsiFile;
import org.jetbrains.annotations.NotNull;

public class BlocContext extends TemplateContextType {
    protected BlocContext() {
        super("Bloc", "Bloc");
    }

    @Override
    public boolean isInContext(@NotNull PsiFile file, int offset) {
        return file.getName().endsWith(".dart");
    }
}

File: extensions/intellij/intellij_generator_plugin/src/main/java/com/bloc/intellij_generator_plugin/live_templates/BlocContext.kt
package com.bloc.intellij_generator_plugin.live_templates

import com.intellij.codeInsight.template.TemplateActionContext
import com.intellij.codeInsight.template.TemplateContextType

class BlocContext : TemplateContextType("Bloc") {
    override fun isInContext(templateActionContext: TemplateActionContext): Boolean {
        return templateActionContext.file.name.endsWith(".dart")
    }
}